plugins/modules/files/iso_customize.py

```diff
@@ -0,0 +1,331 @@
+
+PYCDLIB_IMP_ERR = None
+try:
+    import pycdlib
+    HAS_PYCDLIB = True
+except ImportError:
+    PYCDLIB_IMP_ERR = traceback.format_exc()
+    HAS_PYCDLIB = False
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+from ansible.module_utils.common.text.converters import to_native
+
+MODULE_ISO_CUSTOMIZE = None
+ISO_MODE = "iso9660"
+
+
+# The upper dir exist, we only add subdirectoy
+def iso_add_dir(opened_iso, dir_path):
+    parent_dir, check_dirname = dir_path.rsplit("/", 1)
+    if not parent_dir.strip():
+        parent_dir = "/"
+    check_dirname = check_dirname.strip()
+
+    for dirname, dirlist, dummy_filelist in opened_iso.walk(iso_path=parent_dir.upper()):
+        if dirname == parent_dir.upper():
+            if check_dirname.upper() in dirlist:
+                return 0, ""
+
+            if parent_dir == "/":
+                current_dirpath = "/%s" % check_dirname
+            else:
+                current_dirpath = "%s/%s" % (parent_dir, check_dirname)
+
+            current_dirpath_upper = current_dirpath.upper()
+            try:
+                if ISO_MODE == "iso9660":
+                    opened_iso.add_directory(current_dirpath_upper)
+                elif ISO_MODE == "rr":
+                    opened_iso.add_directory(current_dirpath_upper, rr_name=check_dirname)
+                elif ISO_MODE == "joliet":
+                    opened_iso.add_directory(current_dirpath_upper, joliet_path=current_dirpath)
+                elif ISO_MODE == "udf":
+                    opened_iso.add_directory(current_dirpath_upper, udf_path=current_dirpath)
+            except Exception as err:
+                msg = "Failed to create dir %s with error: %s" % (current_dirpath, to_native(err))
+                return -1, msg
+
+    return 0, ""
+
+
+def iso_add_dirs(opened_iso, dir_path):
+    dirnames = dir_path.strip().split("/")
+
+    current_dirpath = "/"
+    for item in dirnames:
+        if not item.strip():
+            continue
+        if current_dirpath == "/":
+            current_dirpath = "/%s" % item
+        else:
+            current_dirpath = "%s/%s" % (current_dirpath, item)
+
+        ret, ret_msg = iso_add_dir(opened_iso, current_dirpath)
+        if ret != 0:
+            return ret, ret_msg
+
+    return 0, ""
+
+
+def iso_add_file(opened_iso, src_file, dest_file):
+    dest_file = dest_file.strip()
+    if dest_file[0] != "/":
+        dest_file = "/%s" % dest_file
+
+    file_local = src_file.strip()
+
+    file_dir = os.path.dirname(dest_file).strip()
+    file_name = os.path.basename(dest_file)
+    if '.' not in file_name:
+        file_in_iso_path = dest_file.upper() + '.;1'
+    else:
+        file_in_iso_path = dest_file.upper() + ';1'
+
+    if file_dir and file_dir != "/":
+        ret, ret_msg = iso_add_dirs(opened_iso, file_dir)
+        if ret != 0:
+            MODULE_ISO_CUSTOMIZE.fail_json(msg=ret_msg)
+
+    try:
+        # The file will be replaced if it exists in ISO
```

Comment:
If the file will be replaced when it already exists, why are we recommending the user delete the file first then add it? Wouldn't it be simpler to just add (and that will replace) the file?

Reply:
I remove the comment because it's not exactly correct.  Thanks